app/src/main/java/com/example/drawingactivity/MainActivity.kt

```diff
@@ -17,11 +25,83 @@ class MainActivity : AppCompatActivity() {
         viewModel.currentBitmap?.let { drawableArea.setBitmap(it) }
         val colorSelector = findViewById<PrimaryColorCircleView>(R.id.colorCircle)
         colorSelector.onChangeColor = { viewModel.cyclePrimaryColor() }
+        findViewById<Button>(R.id.add_layer).setOnClickListener {
+            colorPickerDialog()
+        }
+
+        if (viewModel.isDialogOpen) {
+            colorPickerDialog()
+        }
+
+        viewModel.onAddLayerSuccess = {
+            Toast.makeText(this, getString(R.string.addLayerSuccess) + it, Toast.LENGTH_SHORT)
```

Comment:
This should be a formatted string. Look up how to do this. You're going to get string with arguments to format it

Reply:
Reformatted String